module_utils/configuration.py

```diff
@@ -403,7 +406,8 @@ def edit_object(self, operation_name, params):
             elif equal_objects(existing_object, data):
                 return existing_object
 
-        return self.send_general_request(operation_name, params)
+        new_object = self.send_general_request(operation_name, params)
```

Comment:
Let's add some test cases in `test_configuration.py` covering scenarios with 204.